### PR TITLE
RDM-13223 

### PR DESCRIPTION
**Pull request by padmapriyanalam**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-13223

### Change description ###

AccessControl for Case Read, Update or Deletion

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/test/resources/mappings/bookcase-definition.json [hunk 1]
```diff
@@ -50,7 +50,94 @@
...
+          "role": "basic-profile",
+          "create": false,
+          "read": true,
+          "update": false,
+          "delete": false
+        }
+      ],
+      "roleToAccessProfiles": [
+        {
+          "case_type_id": "TestAddressBookCase",
```
Comment (markdathornehmcts) [inline src/test/resources/mappings/bookcase-definition.json:64]: Deliberate duplicates?

File: src/test/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControlTest.java [hunk 2]
```diff
@@ -31,6 +32,8 @@
 import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignmentAttributes;
 import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignments;
 import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.enums.GrantType;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.matcher.MatcherType;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.matcher.MatcherType;
```
Comment (markdathornehmcts) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControlTest.java:36]: Duplicate not needed.

File: src/test/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControlTest.java [hunk 3]
```diff
@@ -42,6 +45,8 @@
 import uk.gov.hmcts.ccd.domain.service.common.CaseTypeService;
 import uk.gov.hmcts.ccd.endpoint.exceptions.DownstreamIssueException;
 import uk.gov.hmcts.ccd.infrastructure.user.UserAuthorisation;
+import uk.gov.hmcts.reform.ccd.document.am.model.Classification;
+import uk.gov.hmcts.reform.ccd.document.am.model.Classification;
```
Comment (markdathornehmcts) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControlTest.java:49]: Duplicate not needed

**Review by markdathornehmcts — APPROVED** [review]
A few duplicates to be removed, otherwise all good.

File: src/test/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControlTest.java (hunk 3 above)
Comment (padmapriyanalam) [reply src/test/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControlTest.java]: removed duplicates

**Pull request merged**